lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/completion/MavenCompletionParticipant.java

```diff
@@ -624,64 +624,82 @@ private void internalCollectRemoteGAVCompletion(ICompletionRequest request, bool
 				Collections.singletonList(RemoteRepositoryIndexSearcher.CENTRAL_REPO.getUrl());
 		Set<CompletionItem> updateItems = Collections.synchronizedSet(new HashSet<>(remoteArtifactRepositories.size()));
 		plugin.getIndexSearcher().ifPresent(indexSearcher -> {
-			try {
-				CompletableFuture.allOf(remoteArtifactRepositories.stream().map(repository -> {
-					final CompletionItem updatingItem = new CompletionItem("Updating index for " + repository);
-					updatingItem.setPreselect(true);
-					updatingItem.setInsertText("");
-					updatingItem.setKind(CompletionItemKind.Event);
-					updateItems.add(updatingItem);
-					return indexSearcher.getIndexingContext(URI.create(repository)).thenApplyAsync(index -> {
-						switch (node.getLocalName()) {
-						case GROUP_ID_ELT:
-							// TODO: just pass only plugins boolean, and make getGroupId's accept a boolean
-							// parameter
-							if (onlyPlugins) {
-								indexSearcher.getPluginGroupIds(artifactToSearch, index).stream()
-										.map(groupId -> toCompletionItem(groupId, null, range))
-										.forEach(nonArtifactCollector::addCompletionItem);
-							} else {
-								indexSearcher.getGroupIds(artifactToSearch, index).stream()
-										.map(groupId -> toCompletionItem(groupId, null, range))
-										.forEach(nonArtifactCollector::addCompletionItem);
-							}
-							return updatingItem;
-						case ARTIFACT_ID_ELT:
-							if (onlyPlugins) {
-								artifactInfosCollector
-										.addAll(indexSearcher.getPluginArtifacts(artifactToSearch, index));
-							} else {
+			if (indexSearcher!= null) {
```

Comment:
What's actually changed here in the context of the dependency resolution? We should avoid unrelated changes to ease review and improve traceability.

Reply:
The problem is that if the Central Maven repository us disabled a number of NPE is being reported during the tests which may make some tests to fail. So it was a try to prevent the appearing of those exceptions.

This code is not directly related to the issue and can be removed (or moved to another PR, if acceptable)

Reply:
Having this in a separate change/PR would be great.

Reply:
OK. I'll move it to a new commit